File: AlphabetOrderDiv2/solve/cpp/AlphabetOrderDiv2Solver.cpp
#include <fstream>
#include <topcoder>
#include "AlphabetOrderDiv2.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    vector<string> words; tc::read(ifs, words);
    ifs.close();

    std::ofstream ofs(argv[2]);
    AlphabetOrderDiv2 solver;
    tc::show(ofs, solver.isOrdered(words));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}

File: AlphabetOrderDiv2/solve/java/AlphabetOrderDiv2Solver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class AlphabetOrderDiv2Solver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        List<String> wordsBoxed = (List<String>) reader.next(new TypeRef<List<String>>(){}.getType());
        String[] words = new String[wordsBoxed.size()];
        for (int _i = 0; _i < wordsBoxed.size(); ++_i)
            words[_i] = wordsBoxed.get(_i);
        reader.close();

        AlphabetOrderDiv2 solver = new AlphabetOrderDiv2();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.isOrdered(words));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }
}
